controller/src/main/java/org/jboss/as/controller/ValidateModelStepHandler.java

```diff
@@ -147,6 +150,62 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
 
     }
 
+    private void handleObjectAttributes(ModelNode model, AttributeDefinition attr, String absoluteParentName) throws OperationFailedException {
+        if (attr instanceof ObjectTypeAttributeDefinition) {
+            validateNestedAttributes(model.get(attr.getName()), attr, absoluteParentName);
+        } else if (attr instanceof ObjectListAttributeDefinition) {
+            AttributeDefinition valueType = ((ObjectListAttributeDefinition) attr).getValueType();
+            ModelNode list = model.get(attr.getName());
+            for (int i = 0; i < list.asInt(); i++) {
+                validateNestedAttributes(list.get(i), valueType, absoluteParentName + "[" + i + "]");
```

Comment:
It's possible that list.get(i) is undefined, in which case the validateNestedAttributes call should not be made.

This is unlikely but might as well be robust.